docs/proposals/check-design.md

```diff
@@ -0,0 +1,114 @@
+# oras check draft design
+
+This is a draft design for the `oras check` command.
```

Comment:
could you clarify what content will be checked?

Reply:
For each image specified by the user (by name or digest), we will check:

[for the manifest]
1. If the manifest referenced by the tag or digest exists in the repository.
2. If the manifest's size, digest and media type match the information saved on the repo server. (technically, that means if the descriptor resolved by the repo server matches the actual manifest)

[for each blob, including config and layers]
1. If the blob referenced by the manifest exists in the repository.
2. If the blob's size and digest match the descriptors in the manifest

[check the subject if present, recursively]
1. check the subject's manifest as mentioned above
2. check the subject's layer as mentioned above

Reply:
Maybe just add this text  here?